client/db/src/bench.rs

```diff
@@ -132,7 +135,13 @@ impl<B: BlockT> StateBackend<HashFor<B>> for BenchmarkingState<B> {
 	type TrieBackendStorage = <DbState<B> as StateBackend<HashFor<B>>>::TrieBackendStorage;
 
 	fn storage(&self, key: &[u8]) -> Result<Option<Vec<u8>>, Self::Error> {
-		self.state.borrow().as_ref().ok_or_else(state_err)?.storage(key)
+		let state = self.state.borrow_mut();
+		let db_state = state.as_ref().ok_or_else(state_err)?;
+		let _ = self.usage_info.try_borrow_mut().map(|mut usage_info| {
+			usage_info.reads.ops += 1;
+			usage_info.reads.bytes += key.len() as u64;
```

Comment:
This is NOT good, and I would appreciate some help on how to properly track those reads

Reply:
What you mean? What kind of help do you need?

Reply:
Is the number of bytes read exactly equivalent to `key.len()`?
Is this counting exhaustive? (i.e. are there other places in the code where we should be incrementing those counters?)

Basically I just place this as a placeholder hoping that someone more knowledgeable than me would provide some guidelines as to how / where I could measure those metrics :)

Reply:
You need to add there the number of bytes of the value to the `reads.bytes`. You are not reading the key here.

Reply:
Yeah I was kind of copying some code that I found elsewhere: [lib.rs](https://github.com/paritytech/substrate/blob/c50faf2395218e644859611d703d9fe3a4876f5b/client/db/src/lib.rs#L1127), [stats.rs](https://github.com/paritytech/substrate/blob/c50faf2395218e644859611d703d9fe3a4876f5b/client/db/src/stats.rs#L58).

From what I understand I can't really tell how many bytes I will read from this function?

Reply:
Maybe @NikVolf could help here? :D

Reply:
What you copied from lib.rs is a deletion operation, and we track key length for deletion operation.

For read operation it is value length that should be counted, not key length (for example, what you can observe in `tally_key_read`)

Reply:
Thanks for your help! I've added a commit that does just that! :)